feat: Utilise create-discord-bot (#10013)
discordjs/discord.js: apps/guide/src/content/03-topics/05-intents.mdx

---
title: Intents
category: Topics
---

# Intents

> Intents are an important part of establishing a WebSocket connection, as they define behaviour regarding gateway events and impact received data via the REST API.

Reviewer: ```suggestion
Intents are an important part of establishing a WebSocket connection, as they define behavior regarding gateway events and impact received data via the REST API.
```

## Usage

```js
import { Client, GatewayIntentBits } from 'discord.js';

const client = new Client({
	intents: [GatewayIntentBits.Guilds],
});
```

> This is the most basic usage of intents for discord.js. By specifying _`GatewayIntentBits.Guilds`_, your bot will receive gateway events regarding guilds. This includes receiving initial information about guilds it is in at startup, such as role data and the shard id to be used.

Reviewer 2: sharding has nothing to do with intents! its a separate thing, unrelated to intents. Guild#shardId is a custom prop by us 👀

Author: Yes, that's the point. This is a discord.js guide, not a Discord guide.

Reviewer 2: `and the shard id to be used` is incorrect tho, this intent (or any for that matter) have nothing to do with this

Author: Resolved with 93eed36d2d2e0c47fece9c6064736ef875e61f6c.

> This is the most basic usage of intents for discord.js. By specifying _`GatewayIntentBits.Guilds`_, we will receive gateway events regarding guilds. This includes receiving information about guilds as the bot starts up, such as the roles in the guild and the respective shard id.

Reviewer: ```suggestion
This is the most basic usage of intents for discord.js. By specifying _`GatewayIntentBits.Guilds`_, your bot will receive gateway events regarding guilds. This includes receiving initial information about guilds it is in at startup, such as role data and the shard id to be used.
```

> The full list of _`GatewayIntentBits`_ one can specify can be found <DiscordAPITypesLink type="enum" parent="GatewayIntentBits">on the documentation</DiscordAPITypesLink>. An explanation of what each intent does can be found [on Discord's documentation](https://discord.com/developers/docs/topics/gateway#list-of-intents).

Reviewer: ```suggestion
You can find the full list of _`GatewayIntentBits`_ <DiscordAPITypesLink type="enum" parent="GatewayIntentBits">on the documentation</DiscordAPITypesLink> and an explanation of what each intent does [on Discord's API documentation](https://discord.com/developers/docs/topics/gateway#list-of-intents).
```

## Considerations

In discord.js, some intents require an extra bit of consideration.

### _`GatewayIntentBits.Guilds`_

> discord.js relies heavily on caching in the library. It is recommended to set at least the _`GatewayIntentBits.Guilds`_ intent for discord.js to avoid these pitfalls.

Reviewer: ```suggestion
discord.js relies heavily on caching in the library. We recommend you set at least the _`GatewayIntentBits.Guilds`_ intent to avoid these pitfalls.
```

### _`GatewayIntentBits.GuildMembers`_

Fetching members in a guild via <DocsLink type="class" parent="GuildMemberManager" symbol="fetch" brackets /> requests them over the gateway. As such, this intent is required and you may receive a timeout error if this intent is not specified.

### _`GatewayIntentBits.DirectMessages`_

> This intent is required to receive direct messages. In discord.js however, you **must** specify partials as well. See the partials topic on how this is done.

Reviewer 2: Let's give a direct link to the partials part where it specifies this 🙏

Author: This doesn't exist yet, remember?

### _`GatewayIntentBits.MessageContent`_

Unlike other intents, this only populates user-generated fields. See [Discord's documentation](https://discord.com/developers/docs/topics/gateway#message-content-intent) on what exactly this intent unveils.

It is a common mistake to not see the message content in a message—this is usually because this intent is not specified.

> <Alert title="Info" type="info">

Reviewer 2: so is guild members, so we should either add the same warning there or... idk

Author: Resolved with 5d37673abeb09b5c6d0cf3fb4b01ddd49eefebe1.

	This is a privileged intent. Read on for more information.
</Alert>

## Privileged intents

Some gateway events are considered privileged. Currently, these are:

- _`GatewayIntentBits.GuildPresences`_
- _`GatewayIntentBits.GuildMembers`_
- _`GatewayIntentBits.MessageContent`_

> To use these intents, you will need to enable them in the developer portal. If your bot is in over 75 guilds, you will need to verify your bot and request usage of your desired intents.

Reviewer: ```suggestion
To use these intents, you will need to enable them in the developer portal. If your bot is in over 75 guilds, you will need to verify it and request usage of your desired intents.
```
